### PR TITLE
Feature/co 868 impl

### PR DESCRIPTION
**Pull request by floppyears**

### CODE REVIEW

File: src/main/groovy/edu/oregonstate/mist/positions/resources/PositionsResource.groovy [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    private PositionDAO positionDAO
+
+    PositionsResource(PositionDAO positionDAO) {
+        this.positionDAO = positionDAO
+    }
+
+    @Timed
+    @GET
+    Response getPositions(@QueryParam('businessCenter') String businessCenter) {
+        if (!businessCenter?.trim()) {
```
Comment (gmtcrary) [inline src/main/groovy/edu/oregonstate/mist/positions/resources/PositionsResource.groovy]: Length checking against the businessCenter parameter should be done.

File: src/main/groovy/edu/oregonstate/mist/positions/core/Position.groovy [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import edu.oregonstate.mist.api.jsonapi.ResourceObject
+
+class Position {
+    String title
+    String businessCenter
+    String positionNumber
+    String organizationCode
+
+    ResourceObject toResourceObject() {
+        new ResourceObject(
```
Comment (gmtcrary) [inline src/main/groovy/edu/oregonstate/mist/positions/core/Position.groovy:12]: The self links are currently null/missing.

File: src/main/groovy/edu/oregonstate/mist/positions/resources/PositionsResource.groovy [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    Logger logger = LoggerFactory.getLogger(PositionsResource.class)
+    private PositionDAO positionDAO
+
+    PositionsResource(PositionDAO positionDAO) {
+        this.positionDAO = positionDAO
+    }
+
+    @Timed
+    @GET
+    Response getPositions(@QueryParam('businessCenter') String businessCenter) {
```
Comment (gmtcrary) [inline src/main/groovy/edu/oregonstate/mist/positions/resources/PositionsResource.groovy]: The "type" query parameter should be added as a required parameter as it says in the Swagger. It should be checked for the value "student" as well.

**Review by gmtcrary — CHANGES_REQUESTED** [review]

File: src/main/groovy/edu/oregonstate/mist/positions/resources/PositionsResource.groovy (hunk 3 above)
Comment (floppyears) [reply src/main/groovy/edu/oregonstate/mist/positions/resources/PositionsResource.groovy]: Agreed

**Review by jaredkosanovic — APPROVED** [review]

**Pull request merged**